Sources/SimplenoteSearch/Extensions/NSPredicate+Search.swift

```diff
@@ -74,30 +77,39 @@ extension NSPredicate {
         return NSPredicate(format: "tags MATCHES[n] %@", regex)
     }
 
-    /// Returns a NSPredicate that will match Tags with a given Keyword
+    /// Returns a NSPredicate that will match Tags with a given Query
     ///
     /// -   We'll always analyze the last token in the string
     ///     -   Whenever the last token contains `tag:`, we'll lookup Tags with names containing the payload, excluding exact matches
     ///     -   Otherwise we'll match Tags **containing** the last token
     ///     -   If the `tag:` operator has no actual payload, **every single tag** will be matched (Always True Predicate)
     ///
     @objc
-    public static func predicateForTag(keyword: String) -> NSPredicate {
-        let keywords = keyword.trimmingCharacters(in: .whitespacesAndNewlines).components(separatedBy: .whitespaces)
-        let last = keywords.last?.lowercased() ?? String()
-
-        guard let tag = last.suffix(afterPrefix: lowercasedSearchOperatorForTags) else {
-            return NSPredicate(format: "name CONTAINS[cd] %@", last)
+    public static func predicateForTag(query: SearchQuery) -> NSPredicate {
+        guard let lastItem = query.items.last else {
+            return NSPredicate(value: true)
         }
 
-        guard tag.isEmpty == false else {
-            return NSPredicate(value: true)
+        switch lastItem {
+        case .keyword(let value):
+            return NSPredicate(format: "name CONTAINS[cd] %@", value)
+        case .tag(let value):
+            guard value.isEmpty == false else {
+                return NSPredicate(value: true)
+            }
```

Comment:
This check can be dropped by filtering out empty tags in `parse()`. Note added below!